### PR TITLE
fix/node: include prefixes when calculating routing table hash

### PR DESCRIPTION
**Pull request by Fraser999**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/states/node.rs [hunk 1]
```diff
@@ -2032,8 +2032,9 @@ impl Node {
                      dst: Authority<XorName>)
                      -> Result<(), RoutingError> {
         let sections = self.peer_mgr.pub_ids_by_section();
-        let serialised_sections = serialisation::serialise(&sections)?;
-        if digest == sha256::hash(&serialised_sections) {
+        let prefixes = self.peer_mgr.routing_table().prefixes();
+        let serialised_rt = serialisation::serialise(&(&sections, prefixes))?;
```
Comment (fizyk20) [inline src/states/node.rs]: I'm not sure how serialisation works, but won't `&(&sections, prefixes)` and `&(sections, prefixes)` serialise differently? If yes, we might have to clone `sections` here.

Comment (afck) [inline src/states/node.rs]: Looks like they don't:
https://github.com/rust-lang-nursery/rustc-serialize/blob/master/src/serialize.rs#L1105
But for consistency we could use `&` in both cases anyway.

Comment (fizyk20) [inline src/states/node.rs]: Probably not worth it, then :+1:

**Pull request merged**